wrappers/wrappers/macs2/estimate_d/wrapper.py

```diff
@@ -0,0 +1,65 @@
+import os
+import os.path
+from snakemake import shell
+import tempfile
+
+log = snakemake.log_fmt_shell()
+logfile = None
+extra = snakemake.params.get('extra', '')
+
+label = snakemake.params.block['label']
+extra = snakemake.params.block.get('extra', '')
+
+effective_genome_count = snakemake.params.block.get('effective_genome_count',
+                         snakemake.params.block.get('reference_effective_genome_count', ''))
+
+mfold_lower = snakemake.params.block.get('mfold_lower', '')[0]
+mfold_upper = snakemake.params.block.get('mfold_upper', '')[0]
+
+genome_count_flag = ''
+if effective_genome_count != '':
+    genome_count_flag = ' -g ' + effective_genome_count + ' '
+
+d_estimate_file = snakemake.output.d_estimate
+
+# determine if this parameter set has already been estimated for this file
+found_estimate = False
+if os.path.isfile(d_estimate_file):
+    with open(d_estimate_file) as f:
+        for line in f:
+            parsed = line.split()
+            if len(parsed) == 3 and \
```

Comment:
Are there cases where `d_estimate_file` exists but the estimate is not in that file?

Also, since this `d_estimate_file` is an output for the rule, if it exists when the rule runs but the rule fails for some reason, snakemake will delete it. Is it safe to say that if this rule runs, it should overwrite that file?